core/app/models/spree/shipment.rb

```diff
@@ -386,6 +388,21 @@ def manifest_unstock(item)
       stock_location.unstock item.variant, item.quantity, self
     end
 
+    def resume_inventory_units
+      manifest.each do |item|
+        if item.variant.should_track_inventory?
+          units_available, units_to_back_order = stock_location.fill_status(item.variant, item.quantity)
+
+          inventory_units.canceled.limit(units_available).each(&:on_hand!)
+          inventory_units.canceled.limit(units_to_back_order).each(&:backorder!)
```

Comment:
Is canceling an order the only way to cancel an IU at the moment? Just wondering if other IUs can have the state canceled and they would be resumed, even if not part of the order cancelation.

Reply:
No, you are right, there are other places.
I have to pick only the canceled units for the variant reported on the manifest items.

Reply:
I've used the line_item to be sure.

Reply:
BTW I'm double-checking this to be sure. Sorry for the message spam 🙇 .